extension-manifest-v3/src/utils/engines.js

```diff
@@ -123,15 +125,28 @@ async function loadFromStorage(name) {
 
 async function saveToStorage(name) {
   const engine = loadFromMemory(name);
-  const db = await getDB();
+  const serialized = engine?.serialize();
 
-  const tx = db.transaction('engines', 'readwrite');
-  const table = tx.objectStore('engines');
+  try {
+    const db = await getDB();
 
-  if (engine) {
-    await table.put(engine.serialize(), name);
-  } else {
-    await table.delete(name);
+    const tx = db.transaction('engines', 'readwrite');
+    const table = tx.objectStore('engines');
+
+    if (engine) {
+      await table.put(serialized, name);
+    } else {
+      await table.delete(name);
+    }
+  } catch (e) {
+    if (__PLATFORM__ === 'firefox') {
+      const key = `engines:${name}`;
```

Comment:
Would be worth logging a warning if the fallback is used.

Reply:
I am not sure about it, as it will flood the console, as this is called each time the BG starts.

Reply:
It was a general comment to both places  - We have already logs if the function successfully saves the engine - regardless of the target.